### PR TITLE
Documentation of web-api repository (issue #113)

### LINKED ISSUE #113: Document what the web-api repository is responsible for and how it interact with other repositories
Definition of done:
It should have enough information about the repository that a new student can open this document and gain a clear understanding about why the repository exsist and whats its used for.

### PR DESCRIPTION
**Pull request by Semicolon1071**
Added a description of the purpose of the web-api repository along with its relations to other repositories (issue #113). The entry is made so it is placed between 'Giraf Service' and 'Backend Architecture', along with corresponding reference updates. Further two images has been added, each of which are used in the entry.

### CODE REVIEW

**Comment by mads-bo** [comment]
@aau-giraf/sw611f20 @aau-giraf/sw607f20

File: REST_API_Development/WebApiPurpose.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Web-API purpose
+
+## What is a [Web API](https://en.wikipedia.org/wiki/Web_API)?
+
+A Web API is an application programming interface located on a web-server. It manages the retrieval of requests as well as passing along an associated responses through the HTTP/HTTPS protocol. Such requests are usually in the format of JavaScript Object Notation (Json) or Extensible Markup Language (XML).
```
Comment (Jakobzs) [inline REST_API_Development/WebApiPurpose.md]: "JSON" instead of "Json"

File: REST_API_Development/WebApiPurpose.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Why the repository exists and whats its used for?
+
+The web-API repository is the backend of the [Giraf Project](https://github.com/aau-giraf/web-api), and it is built with the .NET Core in C#. More specifically the Web API is responsible for the handling of all communication coming from and to the API-Client as shown Figure 7.16. If it is a valid request this leads to an [endpoint](./EndpointsAndControllers.md) and if not then an error response is returned. Most request would require the web-API to communicate with the [database](./Database.md) that stores all the data related to the Giraf Project, which allows several of components to be communicating information to the common database.
```
Comment (Jakobzs) [inline REST_API_Development/WebApiPurpose.md]: "and it is built with the .NET Core in C#", instead: "and it is programmed in C# and uses .NET Core 2.2."

File: REST_API_Development/WebApiPurpose.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- DepartmentController
+- PictogramController
+- WeekController
+- WeekTemplateController
+- StatusController
+- ErrorController
+
+![Web-API Controllers](./images/Fig-7-15.png)
+
+# Related Artifacts
```
Comment (Jakobzs) [inline REST_API_Development/WebApiPurpose.md:29]: Related artifacts is empty?

File: REST_API_Development/WebApiPurpose.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## web-api-dotnetcore-build
+Is a Docker containers for the development branch of the webAPI.
```
Comment (Jakobzs) [inline REST_API_Development/WebApiPurpose.md]: "container" instead of "containers"?

File: REST_API_Development/WebApiPurpose.md (hunk 2 above)
Comment (Jakobzs) [inline REST_API_Development/WebApiPurpose.md]: "Giraf" should always be fully uppercase: "GIRAF"

File: REST_API_Development/WebApiPurpose.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+![Web-API in System Overview](./images/Fig-7-16.png)
+
+## Endpoints and Controllers
+
+An endpoint is one end of a communication channel, used to interact with other systems and which usually exposes a set of operations related to a common domain, such as a user or account. When an API interacts with another system, the touchpoints of such communication is called endpoint. An API works by receiving requests and returning appropriate responses. Each endpoint of the web-API is represented by a controller, for which each request is routed to the appropriate controller, e.g. a request for the user endpoint, denoted by the route ```v1/UserController```, would be handled by the UserController. The response returned by the endpoint are mitigated by the appropriate Controller. The current Controllers of the project are:
```
Comment (Simkortet) [inline REST_API_Development/WebApiPurpose.md]: "is called endpoint" -> "is called endpoints".

File: REST_API_Development/WebApiPurpose.md (hunk 3 above)
Comment (Semicolon1071) [reply REST_API_Development/WebApiPurpose.md:29]: The purpose was to have a header for the following few items (web-api-dotnetcore-build and api-client), but if it seems confusing, should it maybe just be removed then? @Seraph2

File: REST_API_Development/WebApiPurpose.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+# Web-API purpose
+
+## What is a [Web API](https://en.wikipedia.org/wiki/Web_API)?
```
Comment (Torpstitution) [inline REST_API_Development/WebApiPurpose.md:3]: Skal der være link her?

**Comment by mathiasandresen** [comment]
Tror tanken er at der skal stå noget i denne fil også: https://github.com/aau-giraf/wiki/tree/master/Repositories/repository_discription.md

I kan jo med fordel linke til anden dokumentation

**Review by Torpstitution — DISMISSED** [review]

File: REST_API_Development/WebApiPurpose.md (hunk 3 above)
Comment (Jakobzs) [reply REST_API_Development/WebApiPurpose.md:29]: I see then. The header makes sense in my opinion.

File: Repositories/repository_discription.md [hunk 7]
```diff
@@ -6,10 +6,10 @@ There are many different repositories in the GIRAF project, but most of them are
 These repositories are those that as of 2019 are active in the Giraf project. 
 
 ### web-api
-The backend Restfull API for the Giraf Project. The API is a .net-core project written in C#. The wiki should be updated so it always reflect the current status of the other repositories as well as other relevant artifacts. 
+The repository contains the backend for the Giraf Project, namely a RESTful API using .NET Core and written in C#. The contents of the repository is further detailed in [REST API Introduction](../REST_API_Development/BackendArchitecture.md), with the specific purpose of the backend described in [Web-API purpose](../REST_API_Development/WebApiPurpose.md).
```
Comment (Jakobzs) [inline Repositories/repository_discription.md]: Missing uppercase "GIRAF".

**Review by Jakobzs — DISMISSED** [review]

File: Repositories/repository_discription.md (hunk 7 above)
Comment (Semicolon1071) [reply Repositories/repository_discription.md]: Done

**Review by Jakobzs — APPROVED** [review]

**Review by Simkortet — APPROVED** [review]

**Pull request merged**